utils/check_inits.py

```diff
@@ -83,9 +87,19 @@ def parse_init(init_file):
     while not lines[line_index].startswith("if TYPE_CHECKING"):
         # If the line is an if is_backend_available, we grab all objects associated.
         backend = find_backend(lines[line_index])
+        # Check if the backend declaration is inside a try block:
+        if _re_try.search(lines[line_index - 1]) is None:
+            backend = None
+
```

Comment:
This actually makes the check not check anything anymore since all the lines with an `if is_xxx_available` are now in a `Try` block.

Reply:
Are you sure? The logic is:

1. A backend has been found (`is_xxx_available` is somewhere on this line)
2. We check the line above for occurrence of `try`.
3. If `try` isn't there then we aren't in a try/except/else block.
4. If we aren't in a try/except/block then we may skip since we're checking for correct _imports_ and all imports happen in try/except/else blocks.

I also empirically checked that commenting out an import in `__init__.py` triggers the appropriate error - so I think this may be working as intended. The main reason for adding this check is because `is_xxx_available` is also used as a condition for logger warnings ("None of PyTorch, TF etc have been found.")